src/backend/utils/adt/datetime.c

```diff
@@ -1972,6 +1973,12 @@ DecodeTimeOnly(char **field, int *ftype, int nf,
 					if (dterr)
 						return dterr;
 				}
+				/*
+				 * For date format like 'yyyy-mm-dd', we should return time 00:00:00.0000000
+				 * instead of marking it as invalid time format
+				 */
+				else if (sql_dialect == SQL_DIALECT_TSQL && nf == 1 && ftype[nf - 1] == DTK_DATE)
+					return 0;
 				/* otherwise, this is a time and/or time zone */
```

Comment:
nf = numField (parsed and calculated based on delimiter ' ' and . in any input date)
Ex.
'08-04-1978 13:56:32.125431' has nf count = 3 (08-04-1978, 13:56:32, 125431)
'08-04-1978 13:56:32' has nf count = 2 (08-04-1978, 13:56:32)
'08-04-1978' has nf count = 1 (08-04-1978)